Abstractions project

MattEqualsCoder opened the pull request:
Created a new abstractions project to hold interfaces and whatnot. For now, I just moved some of the Tracker stuff in there. The game mode interface will go in there, and we can probably slowly pull other things in there over time.

Vivelin reviewed (approved):
Looks good. I do wonder if some of these new interfaces should be abstract classes instead.

MattEqualsCoder commented:
> Looks good. I do wonder if some of these new interfaces should be abstract classes instead.

I guess with as many properties as there are here, I suppose abstract classes would make more sense, yeah.

MattEqualsCoder commented:
Switched to abstract classes for AutoTracker, Tracker, and the SpeechRecognitionService (due to the event - interfaces don't seem to work well with events that aren't used).

While I was at it, I cleaned up a lot of the warnings that had cropped up in builds since the .net 7 upgrade, most of which were in regards to code being only accessible on Windows.
